microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/core/utils/AnnotationUtils.java

```diff
@@ -183,7 +184,8 @@ private static boolean isMatchAnnotationFullyQualifiedName(IAnnotation annotatio
 	 *         false otherwise.
 	 */
 	public static boolean isMatchAnnotation(Annotation annotation, String annotationName) {
-		return annotationName.endsWith(annotation.getTypeName().getFullyQualifiedName());
+		IAnnotationBinding binding = annotation.resolveAnnotationBinding();
+		return annotationName.endsWith(binding.getAnnotationType().getQualifiedName());

```

Comment:
You can even do : `annotation.resolveTypeBinding().getQualifiedName()` though I do like your approach because it goes more to what the type happens to be.

Should we guard off `binding.getAnnotationType()` ? If I look into the implementation, it can return `null`.

Reply:
Sure, I'll add a check for that